### PR TITLE
Docs: Improve Conda docs - PyPI + lock files

### PR DESCRIPTION
**Pull request by ewels**
* Add docs about Conda lock files
* Create new headings specifically for PyPI, text files and lock files
* Add mention of ability to use PyPI packages to the Wave docs, with link

@netlify /conda#use-conda-environment-files

### CODE REVIEW

File: docs/conda.md [hunk 1]
```diff
@@ -110,7 +97,26 @@ process foo {
...
+  - pip
+  - pip:
+    - numpy
+    - pandas
+    - matplotlib
+```
+
+### Conda text files
+
+It is possible to provide the dependencies using a plain text file, just listing each package name as a separate line. For example:
```
Comment (christopher-hakkaart) [inline docs/conda.md:119]:
```suggestion
It is possible to provide dependencies by listing each package name as a separate line in a plain text file. For example:
```

File: docs/conda.md [hunk 2]
```diff
@@ -122,6 +128,43 @@ bioconda::multiqc=1.4
...
+conda list --explicit > spec-file.txt
+```
+
+or if using Mamba / Micromamba:
+
+```bash
+micromamba env export --explicit > spec-file.txt
+```
+
+They can also be downloaded from Wave build pages.
```
Comment (christopher-hakkaart) [inline docs/conda.md:147]:
```suggestion
Conda lock files can also be downloaded from Wave build pages.
```

File: docs/conda.md [hunk 3]
```diff
@@ -122,6 +128,43 @@ bioconda::multiqc=1.4
...
+
+These files include the every package, including dependencies. As such, no conda environment resolution step is needed. This is faster and also better for reproducibility between runs.
```
Comment (christopher-hakkaart) [inline docs/conda.md:149]:
```suggestion
These files include every package and their dependencies. As such, no Conda environment resolution step is needed. This is faster and more reproducible.
```

File: docs/conda.md [hunk 4]
```diff
@@ -122,6 +128,43 @@ bioconda::multiqc=1.4
 Like before, the extension matters. Make sure the dependencies file has a `.txt` extension.
 :::
 
+### Conda lock files
+
+The final way to provide packages to Conda is with [Conda lock files](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#identical-conda-envs).
+
+These are generated from existing conda environements using the following command:
```
Comment (christopher-hakkaart) [inline docs/conda.md:135]:
```suggestion
These are generated from existing Conda environments using the following command:
```

File: docs/conda.md [hunk 5]
```diff
@@ -122,6 +128,43 @@ bioconda::multiqc=1.4
...
+
+```bash
+micromamba env export --explicit > spec-file.txt
+```
+
+They can also be downloaded from Wave build pages.
+
+These files include the every package, including dependencies. As such, no conda environment resolution step is needed. This is faster and also better for reproducibility between runs.
+
+The files contain package URLs and optionally also a md5hash for each download to confirm identity:
```
Comment (christopher-hakkaart) [inline docs/conda.md:151]:
```suggestion
The files contain package URLs and an optional md5hash for each download to confirm identity:
```

File: docs/wave.md [hunk 6]
```diff
@@ -90,6 +90,8 @@ conda.channels = 'conda-forge,bioconda'
 ```
 :::
 
+Packages from the [Python Package Index](https://pypi.org/) can also be used by using a Conda `environment.yml` file - see the documentation about {ref}`Conda and PyPI <conda-pypi>` for details.
```
Comment (christopher-hakkaart) [inline docs/wave.md:93]:
```suggestion
Packages from the [Python Package Index](https://pypi.org/) can also be added to a Conda `environment.yml` file. See {ref}`Conda and PyPI <conda-pypi>` for more information.
```

**Review by christopher-hakkaart — CHANGES_REQUESTED** [review]

File: docs/conda.md [hunk 7]
```diff
@@ -110,7 +97,26 @@ process foo {
 The environment file name **must** have a `.yml` or `.yaml` extension or else it won't be properly recognised.
 :::
 
-Alternatively, it is possible to provide the dependencies using a plain text file, just listing each package name as a separate line. For example:
+(conda-pypi)=
+### Python Packages from PyPI
+
+Conda environment files can also be used to install Python packages from the [PyPI repository](https://pypi.org/)), through the `pip` package manager (which must also be explicitly listed as a required package):
```
Comment (pditommaso) [inline docs/conda.md:103]:
```suggestion
Conda environment files can also be used to install Python packages from the [PyPI repository](https://pypi.org/), through the `pip` package manager (which must also be explicitly listed as a required package):
```

File: docs/conda.md [hunk 8]
```diff
@@ -122,6 +128,43 @@ bioconda::multiqc=1.4
...
+conda list --explicit > spec-file.txt
+```
+
+or if using Mamba / Micromamba:
+
+```bash
+micromamba env export --explicit > spec-file.txt
+```
+
+Conda lock files can also be downloaded from Wave build pages.
```
Comment (pditommaso) [inline docs/conda.md:147]:
```suggestion
Conda lock files can also be downloaded from [Wave](https://seqera.io/wave/) build pages.
```

**Review by pditommaso — APPROVED** [review]
well done

**Pull request merged**